tests/testthat/test-shinytest2-init.R

```diff
@@ -56,3 +56,58 @@ testthat::test_that("e2e: init creates UI containing specified title, favicon, h
   )
   app$stop()
 })
+
+testthat::test_that("e2e: teal app initializes with Show R Code modal", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = example_module(label = "Example Module")
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check if button exists.
+  button_selector <- sprintf("#%s-rcode-button", app$active_module_ns())
+  testthat::expect_equal(
+    app$get_text(button_selector),
+    "Show R code"
+  )
+
+  app$click(selector = button_selector)
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check header and title content.
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-header > h4"),
+    "Example Code"
+  )
+  testthat::expect_equal(
+    app$get_text(sprintf("#%s-rcode-copy_button1", app$active_module_ns())),
+    "Copy to Clipboard"
+  )
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-body > div > button:nth-child(2)"),
+    "Dismiss"
+  )
```

Comment:
I think those buttons are duplicated, since they exist in the footer already : /
Raised an issue to fix that in here https://github.com/insightsengineering/teal.widgets/issues/233

Reply:
that is actually a feature:) https://github.com/insightsengineering/teal.widgets/issues/233#issuecomment-1994511156

Reply:
lol